plugins/modules/mso_schema_template_external_epg_selector.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Cindy Zhao (@cizhao) <cizhao@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mso_schema_template_external_epg_selector
+short_description: Manage External EPG selector in schema templates
+description:
+- Manage External EPG selector in schema templates on Cisco ACI Multi-Site.
+author:
+- Cindy Zhao (@cizhao)
+options:
+  schema:
+    description:
+    - The name of the schema.
+    type: str
+    required: yes
+  template:
+    description:
+    - The name of the template to change.
+    type: str
+    required: yes
+  external_epg:
+    description:
+    - The name of the External EPG to be managed.
+    type: str
+    required: yes
+  selector:
+    description:
+    - The name of the selector.
+    type: str
+  expressions:
+    description:
+    - Expressions associated to this selector.
+    type: list
+    suboptions:
+      type:
+        description:
+        - The name of the expression which in this case is always IP address.
+        required: true
+        type: str
+        aliases: [ IP ]
+      operator:
+        description:
+        - The operator associated with the expression which in this case is always equals.
+        required: true
+        type: str
+      value:
+        description:
+        - The value of the IP Address associated with the expression.
+        required: true
+        type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+seealso:
+- module: mso_schema_template_externalepg
+extends_documentation_fragment: cisco.mso.modules
+'''
+
+EXAMPLES = r'''
+- name: Add a selector to an External EPG
+  cisco.mso.mso_schema_template_anp_epg_selector:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    external_epg: extEPG 1
+    selector: selector_1
+    expressions:
+      - type: ipAddress
```

Comment:
this should just be a choice and either choice is 'ip_address' or just 'ip' but we should comply with snake case and change it in the logic before putting it in payload.